src/main/java/BMO/Parser.java

```diff
@@ -0,0 +1,104 @@
+package bmo;
+
+/**
+ * Parser class that parses user input and returns an array of strings containing the command and its arguments.
+ */
+public class Parser {
+    /**
+     * Parses the user input and returns an array of strings containing the command and its arguments.
+     *
+     * @param userInput The user input to be parsed.
+     * @return An array of strings containing the command at index 0 and its arguments.
+     * @throws BmoException If the user input is invalid or missing arguments.
+     */
+    public String[] parse(String userInput) throws BmoException {
+        String[] parsedInput = new String[5];
```

Comment:
im curious, are there any reasons why an array of length 5 was created

Reply:
The command with the most number of arguments would be creating an Event task. It has 4 arguments which are the event command, task description, start date, end date. Each argument takes up an index in the parsedInput array. However, you are right, index 4 of the array is never used. I shall change the size of the array to length 4.